examples/deferred_read_csv.py

```diff
@@ -0,0 +1,49 @@
+import letsql as ls
+from letsql.common.utils.defer_utils import (
+    deferred_read_csv,
```

Comment:
Should we add `deferred_read_csv` as an api function?

Reply:
we can, though i would like it to be `read_csv` with "deferred" implied

related, we should not introspect on backends to detect what the csv/parquet source is and instead require users to use `deferred_read_{csv,parquet}` (or eventual api function `read_{csv,parquet}`) if they want to be able to cache the expr